### ISSUE TITLE
can't get verts, features and faces list from marching_tetrahedra function on own data

### ISSUE THREAD

**Issue by zxy0304** [YuelangX/Gaussian-Head-Avatar#23]
hi yuelang!
I tried on my own data, followed the instruction in 3DMM and seccussfully fitted the model. But when process the training process, This line returned none. 

```verts_list, features_list, faces_list = marching_tetrahedra(verts_deformed, features, self.tets, sdf)```

More specifically, inside of this function, all the vertices seems to fail the occupancy sum test, occ_sum is None.

``` with torch.no_grad():
occ_n = sdf > 0
occ_fx4 = occ_n[tets.reshape(-1)].reshape(-1, 4)
occ_sum = torch.sum(occ_fx4, -1)
valid_tets = (occ_sum > 0) & (occ_sum < 4)
occ_sum = occ_sum[valid_tets]
```

Have you also encountered this problem by any chance or do you have any thoughts and comments about how to fix it? I don't think this is a problem from 3dmm fitthing by the way, because the visualize result looks correct to me.

![image_0231](https://github.com/YuelangX/Gaussian-Head-Avatar/assets/119600913/8d56a594-737a-407e-bc28-5fc57398cd01)

thx a lot!

**Comment by YuelangX** [comment]
Did you initialize the neutral mesh as a sphere? When did this problem occur? And visualization results?

**Comment by zxy0304** [comment]
> Did you initialize the neutral mesh as a sphere? When did this problem occur? And visualization results?

Yes, I initialized it as a sphere. I encountered the problem when training the first state, in file dmtet_utils.py, line 51. Occ_sum is a Nonetype, since all sdf value > 0. I tried to substract 0.5 for each sdf value since maybe sdf=0.5 is the boundary(? After this modification the code can run now, but I don't think it is the right way to fix it:(

I didn't have any visualize result for training this gaussian head avatar yet, and i have attached the visualization for 3DMM model as above.

**Comment by zxy0304** [comment]
Hi yuelang, thank you for answering!
I can train meshhead now by manually subtract 0.5 for each sdf value, and here is the visualization. However, I encountered another problem when train gaussian head: all visualization images are offsets from the original. Do you know what's the reason? Are there something wrong with the extrinsic parameters of camera? Thank u so much for your help!
![007600](https://github.com/YuelangX/Gaussian-Head-Avatar/assets/119600913/08a43853-06b3-432e-86e8-317bf860439e)
![000300](https://github.com/YuelangX/Gaussian-Head-Avatar/assets/119600913/e5d16011-937b-496b-8f6d-746c2bdcbbd2)

**Comment by YuelangX** [comment]
When reading the data in Dataset, I crop the image and modify the intrinsic in function "CropImage". In addition, I also crop images in GaussianHeadTrainer. You can check whether it is caused by these two parts.

**Comment by ketan-gan** [comment]
Hey @zxy0304, I was wondering if your custom dataset was monocular or multi-view. If it was monocular could you share your experience with training and the output quality? Thanks!
